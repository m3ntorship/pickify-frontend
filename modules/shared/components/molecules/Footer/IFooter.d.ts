declare namespace IFooter {
  export interface IProps {
    numberOfVotes?: number;
    showResult: boolean;
  }
}
export { IFooter };
