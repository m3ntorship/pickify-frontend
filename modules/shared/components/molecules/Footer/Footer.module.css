.footer {
  @apply px-3 bg-transparent flex flex-col justify-between;
}

.list-parent {
  @apply flex flex-wrap items-start p-0 m-0;
}

.link-item {
  @apply mr-3 mt-1 last:mr-0 leading-4;
}

.link {
  @apply hover:underline font-light text-xs text-dark-grey
  whitespace-nowrap;
}

.copy-rights {
  @apply font-light text-xs text-grey;
}
