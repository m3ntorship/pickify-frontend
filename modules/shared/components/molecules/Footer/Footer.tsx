import React from 'react';
import type { FC, ReactElement } from 'react';
import type { IFooter } from './IFooter';
import Share from '../../icons/share.svg';
import styles from './Footer.module.css';

const Footer: FC<IFooter.IProps> = ({
  numberOfVotes,
  showResult = false,
}): ReactElement => {
  return (
    <div className={styles.container}>
      {!showResult && <span> Vote to uncover the total number of voters </span>}
      {showResult && <span>{numberOfVotes} votes</span>}
      <span>
        <Share />
      </span>
    </div>
  );
};

export default Footer;
