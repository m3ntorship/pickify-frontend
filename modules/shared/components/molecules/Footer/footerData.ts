export const footerData = [
  {
    name: 'about',
    path: '/',
    content: 'About',
  },
  { name: 'team', path: '/credits', content: 'Team' },
  { name: 'blog', path: '/', content: 'Blog' },
  { name: 'careers', path: '/', content: 'Careers' },
  {
    name: 'terms-and-conditions',
    path: '/',
    content: 'Terms & Conditions',
  },
  { name: 'help', path: '/', content: 'Help' },
  { name: 'contact', path: '/', content: 'Contact' },
];
