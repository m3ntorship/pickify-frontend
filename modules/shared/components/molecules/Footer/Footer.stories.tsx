import React from 'react';
import type { Story, Meta } from '@storybook/react';
import Footer from './Footer';

export default {
  title: 'Molecules/Footer',
  component: Footer,
} as Meta;

const Template: Story = (args) => {
  return (
    <>
      <div className="mb-8">
        <Footer {...args} />
      </div>
      <div className="w-19xl m-auto">
        <Footer {...args} />
      </div>
    </>
  );
};

export const footer = Template.bind({});
