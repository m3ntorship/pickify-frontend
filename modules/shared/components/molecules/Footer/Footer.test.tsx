import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Footer from './Footer';

describe('Testing Footer with snapshot', () => {
  it('Testing Footer with required props', () => {
    const tree = renderer
      .create(<Footer numberOfVotes={120} showResult={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Testing Footer with required props', () => {
    const tree = renderer
      .create(<Footer numberOfVotes={120} showResult />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
