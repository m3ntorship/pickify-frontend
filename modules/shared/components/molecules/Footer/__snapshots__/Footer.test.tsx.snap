// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Radio with snapshot Testing Radio with required props 1`] = `
<div
  className="container"
>
  <span>
     Vote to uncover the total number of voters 
  </span>
  <span>
    <svg
      data-file-name="SvgShare"
    />
  </span>
</div>
`;

exports[`Testing Radio with snapshot Testing Radio with required props 2`] = `
<div
  className="container"
>
  <span>
    120
     votes
  </span>
  <span>
    <svg
      data-file-name="SvgShare"
    />
  </span>
</div>
`;
